test/metabase/db/data_migrations_test.clj

```diff
@@ -331,3 +334,100 @@
           (testing "And it is idempotent"
             (#'migrations/migrate-click-through)
             (is (= expected-settings (get-settings!)))))))))
+
+(defn- do-with-temporary-setting-value
+  "Temporarily update the value of a setting, execute `thunk`, then re-establish the original value"
+  [setting-k value thunk]
+  (assert (and (string? setting-k) (string? value))
+          (format "key and value should be string, got %s and %s" (type setting-k) (type value)))
+  (let [original-value (db/select-one-field :value Setting :key setting-k)]
+    (try
+     (if original-value
+       (db/update! Setting setting-k :value value)
+       (db/insert! Setting :key setting-k :value value))
+     (thunk)
+     (finally
+       (if original-value
+         (db/update! Setting setting-k :value original-value)
+         (db/delete! Setting :key setting-k))))))
+
+(defmacro ^:private with-temporary-raw-setting-values
+  "Temporarily write settings into Setting table, run body, re-establish the original settings.
+  Note that this will not bind these settings to cache so using `setting/get` might not return the
+  temporary values"
+  [[setting-k value & more :as bindings] & body]
+  (assert (even? (count bindings)) "mismatched setting/value pairs: is each setting name followed by a value?")
+  (if (empty? bindings)
+    `(do ~@body)
+    `(do-with-temporary-setting-value ~(name setting-k) ~value
+       (fn []
+         (with-temporary-raw-setting-values ~more
+           ~@body)))))
+
+(defn- call-with-ldap-and-sso-configured [ldap-group-mapping sso-group-mapping f]
+  (premium-features-test/with-premium-features #{:sso}
+    (with-temporary-raw-setting-values
+      [ldap-enabled                       "true"
+       ldap-host                          "http://localhost:8888"
+       ldap-user-base                     "dc=metabase,dc=com"
+       ldap-group-mappings                (json/generate-string ldap-group-mapping)
+       ldap-sync-admin-group              "false"
+       jwt-enabled                        "true"
+       jwt-identity-provider-uri          "http://test.idp.metabase.com"
+       jwt-shared-secret                  (crypto-random/hex 32)
+       jwt-group-mappings                 (json/generate-string sso-group-mapping)
+       saml-enabled                       "true"
+       saml-identity-provider-uri         "http://test.idp.metabase.com"
+       saml-identity-provider-certificate (slurp "test_resources/sso/auth0-public-idp")
```

Comment:
Is it worth giving this filename an extension? Not sure what the right one is here

Reply:
how about `.cert` tho?